test/libsolidity/util/TestFileParser.cpp

```diff
@@ -77,76 +75,99 @@ vector<solidity::frontend::test::FunctionCall> TestFileParser::parseFunctionCall
 
 				try
 				{
-					if (accept(Token::Library, true))
-					{
-						expect(Token::Colon);
-						call.signature = m_scanner.currentLiteral();
-						expect(Token::Identifier);
-						call.kind = FunctionCall::Kind::Library;
-						call.expectations.failure = false;
-					}
-					else if (accept(Token::Storage, true))
+					if (accept(Token::Gas, true))
 					{
-						expect(Token::Colon);
-						call.expectations.failure = false;
-						call.expectations.result.push_back(Parameter());
-						// empty / non-empty is encoded as false / true
-						if (m_scanner.currentLiteral() == "empty")
-							call.expectations.result.back().rawBytes = bytes(1, uint8_t(false));
-						else if (m_scanner.currentLiteral() == "nonempty")
-							call.expectations.result.back().rawBytes = bytes(1, uint8_t(true));
+						if (calls.empty())
+							throw TestParserError("Expected function call before gas usage filter.");
+
+						string runType = m_scanner.currentLiteral();
+						if (set<string>{"ir", "irOptimized", "legacy", "legacyOptimized"}.count(runType) > 0)
+						{
+							m_scanner.scanNextToken();
+							expect(Token::Colon);
+							calls.back().expectations.gasUsed[runType] = u256(parseDecimalNumber());
+						}
 						else
-							throw TestParserError("Expected \"empty\" or \"nonempty\".");
-						call.kind = FunctionCall::Kind::Storage;
-						m_scanner.scanNextToken();
+							throw TestParserError(
+								"Expected \"ir\", \"irOptimized\", \"legacy\", or \"legacyOptimized\"."
+							);
+
 					}
 					else
 					{
-						bool lowLevelCall = false;
-						tie(call.signature, lowLevelCall) = parseFunctionSignature();
-						if (lowLevelCall)
-							call.kind = FunctionCall::Kind::LowLevel;
+						FunctionCall call;
 
```

Comment:
How do you handle duplicate expectations? E.g.

```c++
// gas ir: 1
// gas legacy: 2
// gas ir: 3
```

Reply:
Good point, will add error in case we have duplicate expectations. So far, I think it would just use second value and pass the test if it is correct...